lib/azuremetadata/azuremetadata.py

```diff
@@ -2,10 +2,38 @@
 
 
 class AzureMetadata:
-    def __init__(self):
-        pass
+    def __init__(self, api_version=None):
```

Comment:
Just set "2017-04-02" as the default then you can drop the inline if condition

Reply:
When `--api` argument is not specified, it's value is `None` and it will override the default value.